plugin/plugin_test.go

```diff
@@ -0,0 +1,383 @@
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package plugin
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd/services"
+)
+
+func RegisterClear() {
+	register.Lock()
+	defer register.Unlock()
+	register.r = nil
+}
+
+func MockPluginFilter(*Registration) bool {
+	return false
+}
+
+var tasksServiceRequires = []Type{
+	RuntimePlugin,
+	RuntimePluginV2,
+	MetadataPlugin,
+	TaskMonitorPlugin,
+}
+
+func TestContainerdPlugin(t *testing.T) {
+	RegisterClear()
+	Register(&Registration{
+		Type: TaskMonitorPlugin,
+		ID:   "cgroups",
+	})
+	Register(&Registration{
+		Type:     ServicePlugin,
+		ID:       services.TasksService,
+		Requires: tasksServiceRequires,
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.IntrospectionService,
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.NamespacesService,
+		Requires: []Type{
+			MetadataPlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "namespaces",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "content",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "containers",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.ContainersService,
+		Requires: []Type{
+			MetadataPlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "events",
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "leases",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.LeasesService,
+		Requires: []Type{
+			MetadataPlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "diff",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.DiffService,
+		Requires: []Type{
+			DiffPlugin,
+		},
+	})
+	Register(&Registration{
+		Type: ServicePlugin,
+		ID:   services.SnapshotsService,
+		Requires: []Type{
+			MetadataPlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "snapshots",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "version",
+	})
+	Register(&Registration{
+		Type: GRPCPlugin,
+		ID:   "images",
+		Requires: []Type{
+			ServicePlugin,
+		},
+	})
+	Register(&Registration{
+		Type: GCPlugin,
```

Comment:
Looks like a typo? `s/GCPlugin/GRPCPlugin/`

Reply:
thanks for review  :)